braindecode/models/eeginception_mi.py

```diff
@@ -0,0 +1,289 @@
+            out_features=self.n_classes,
+            bias=True,
+        )
+
+        self.softmax = nn.LogSoftmax(dim=1)
+
+    def forward(
+        self,
+        X: torch.Tensor,
+    ) -> torch.Tensor:
+        X = self.ensuredims(X)
+        X = self.dimshuffle(X)
+
+        res1 = self.residual_block_1(X)
+
+        out = self.initial_inception_module(X)
+        for layer in self.intermediate_inception_modules_1:
+            out = layer(out)
+
+        out = out + res1
+
+        res2 = self.residual_block_2(out)
+
+        for layer in self.intermediate_inception_modules_2:
+            out = layer(out)
+
+        out = res2 + out
+
+        out = self.ave_pooling(out)
+        out = self.flat(out)
+        out = self.fc(out)
+        return self.softmax(out)
+
+
+class _InceptionModuleMI(nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        n_filters,
+        n_convs,
+        kernel_unit_s=0.1,
+        sfreq=250,
+        activation=nn.ReLU(),
+    ):
+        super().__init__()
+        self.in_channels = in_channels
+        self.n_filters = n_filters
+        self.n_convs = n_convs
+        self.kernel_unit_s = kernel_unit_s
+        self.sfreq = sfreq
+
+        self.bottleneck = nn.Conv2d(
+            in_channels=self.in_channels,
+            out_channels=self.n_filters,
+            kernel_size=1,
+            bias=True,
+        )
+
+        kernel_unit = int(self.kernel_unit_s * self.sfreq)
+
+        # XXX Maxpooling is usually used to reduce spatial resolution, with a
+        # stride equal to the kernel size... But it seems the authors use
+        # stride=1 in their paper according to the output shapes from Table3,
+        # although this is not clearly specified in the paper text.
+        self.pooling = nn.MaxPool2d(
+            kernel_size=(1, kernel_unit),
+            stride=1,
+            padding=(0, int(kernel_unit // 2)),
+        )
+
+        self.pooling_conv = nn.Conv2d(
+            in_channels=self.in_channels,
+            out_channels=self.n_filters,
+            kernel_size=1,
+            bias=True,
+        )
+
+        self.conv_list = nn.ModuleList([
+            nn.Conv2d(
+                in_channels=self.n_filters,
+                out_channels=self.n_filters,
+                kernel_size=(1, (n_units * 2 + 1) * kernel_unit),
+                padding="same",
+                bias=True,
+            ) for n_units in range(self.n_convs)
+        ])
+
+        self.bn = nn.BatchNorm2d(self.n_filters * (self.n_convs + 1))
+
+        self.activation = activation
+
+    def forward(
+        self,
+        X: torch.Tensor,
+    ) -> torch.Tensor:
+        X1 = self.bottleneck(X)
+
+        X1 = [conv(X1) for conv in self.conv_list]
+
+        X2 = self.pooling(X)
+        X2 = self.pooling_conv(X2)
+
+        out = torch.cat(X1 + [X2], 1)
+
+        out = self.bn(out)
+        return self.activation(out)
+
+
+class _ResidualBlockMI(nn.Module):
```

Comment:
I would rather change the name to `_ResidualModuleMI` to fit the description in the paper. Residual block is also often used to describe whole block with the residual connection inside so it can be misleading.

The forward method looks like a standard `nn.Sequential` forward, so we can remove at least the forward implementation by inheriting from `nn.Sequential` or we can remove this class and create `nn.Sequential` objects directly in `EEGInceptionMI.__init__`.

Reply:
Thanks @sliwy ! I just renamed it into `_ResidualModuleMI`, you are right. It is also the way they call it in the paper actually!

Concerning inheriting from `nn.Sequential`, to be honest I am not a big fan and usually prefer to code the `forward` method myself as it gives more flexibility for debugging. Doing it would reduce a bit the number of lines in `_ResidualModuleMI` but without much code clarity improvements IMO. And doing it for `_InceptionModuleMI` and `EEGInceptionMI` could actually degrade clarity I think, as those don't have a straightforward sequential structure. Did I convince you on this point? :)

Reply:
For `_InceptionModuleMI` and `EEGInceptionMI` - no doubts, it should stay as `nn.Module` with custom forward method.

In the case of `_ResidualModuleMI`, I would still use `nn.Sequential` but it's up to you - it depends on taste. 

@cedricrommel great contribution! To me it's ready to merge.